src/main/java/com/uber/cadence/internal/logging/ReplayAwareLogger.java

```diff
@@ -0,0 +1,428 @@
+    if (shouldSkipLogging()) return;
+
+    log.warn(format, arg1, arg2);
+  }
+
+  @Override
+  public void warn(String msg, Throwable t) {
+    if (shouldSkipLogging()) return;
+
+    log.warn(msg, t);
+  }
+
+  @Override
+  public boolean isWarnEnabled(Marker marker) {
+    return log.isWarnEnabled(marker);
+  }
+
+  @Override
+  public void warn(Marker marker, String msg) {
+    if (shouldSkipLogging()) return;
+
+    log.warn(marker, msg);
+  }
+
+  @Override
+  public void warn(Marker marker, String format, Object arg) {
+    if (shouldSkipLogging()) return;
+
+    log.warn(marker, format, arg);
+  }
+
+  @Override
+  public void warn(Marker marker, String format, Object arg1, Object arg2) {
+    if (shouldSkipLogging()) return;
+
+    log.warn(marker, format, arg1, arg2);
+  }
+
+  @Override
+  public void warn(Marker marker, String format, Object... arguments) {
+    if (shouldSkipLogging()) return;
+
+    log.warn(marker, format, arguments);
+  }
+
+  @Override
+  public void warn(Marker marker, String msg, Throwable t) {
+    if (shouldSkipLogging()) return;
+
+    log.warn(marker, msg, t);
+  }
+
+  @Override
+  public boolean isErrorEnabled() {
+    return log.isErrorEnabled();
+  }
+
+  @Override
+  public void error(String msg) {
+    if (shouldSkipLogging()) return;
+
+    log.error(msg);
+  }
+
+  @Override
+  public void error(String format, Object arg) {
+    if (shouldSkipLogging()) return;
+
+    log.error(format, arg);
+  }
+
+  @Override
+  public void error(String format, Object arg1, Object arg2) {
+    if (shouldSkipLogging()) return;
+
+    log.error(format, arg1, arg2);
+  }
+
+  @Override
+  public void error(String format, Object... arguments) {
+    if (shouldSkipLogging()) return;
+
+    log.error(format, arguments);
+  }
+
+  @Override
+  public void error(String msg, Throwable t) {
+    if (shouldSkipLogging()) return;
+
+    log.error(msg, t);
+  }
+
+  @Override
+  public boolean isErrorEnabled(Marker marker) {
+    return log.isErrorEnabled(marker);
+  }
+
+  @Override
+  public void error(Marker marker, String msg) {
+    if (shouldSkipLogging()) return;
+
+    log.error(marker, msg);
+  }
+
+  @Override
+  public void error(Marker marker, String format, Object arg) {
+    if (shouldSkipLogging()) return;
+
+    log.error(marker, format, arg);
+  }
+
+  @Override
+  public void error(Marker marker, String format, Object arg1, Object arg2) {
+    if (shouldSkipLogging()) return;
+
+    log.error(marker, format, arg1, arg2);
+  }
+
+  @Override
+  public void error(Marker marker, String format, Object... arguments) {
+    if (shouldSkipLogging()) return;
+
+    log.error(marker, format, arguments);
+  }
+
+  @Override
+  public void error(Marker marker, String msg, Throwable t) {
+    if (shouldSkipLogging()) return;
+
+    log.error(marker, msg, t);
+  }
+
+  private boolean shouldSkipLogging() {
+    return context.isReplaying() && !enableLoggingInReplay.get();
```

Comment:
I think it is incorrect as it doesn't log when not replaying. Should be
`!context.isReplaying() || (isReplaying && !enableLoggingInReplay())`
Make sure that unit test fails because of that before fixing.

Reply:
if isReplaying is false then should skip is false as well, right?
The current unit test covers this logic. If it fails then we won't see anything in the log :)

Reply:
If is replaying is false then it means that code is executed for the first time and should not skip logging

Reply:
Now I understand. I missed the shouldSkip is negative of shouldLog